src/mmw/js/src/user/controllers.js

```diff
@@ -12,13 +13,14 @@ var SignUpController = {
         }).render();
     },
 
-    itsiSignUp: function(username, first_name, last_name) {
+    itsiSignUp: function(username, first_name, last_name, queryParams) {
         new views.ItsiSignUpModalView({
             app: App,
             model: new models.ItsiSignUpFormModel({
                 username: username,
                 first_name: first_name,
-                last_name: last_name
+                last_name: last_name,
+                next: queryParams ? utils.parseQueryString(queryParams).next : null
```

Comment:
Hmm at first I was thinking that the back-end always has to redirect somewhere, so we default to `/` if no `next` is specified. The front-end on the other hand has the third option of not redirecting at all, which is what the current implementation does here: https://github.com/WikiWatershed/model-my-watershed/pull/578/files#diff-bbc329caa6aad1d9e2bf745db301390fR381. However, now that I think about it, I think we do always want to redirect away from the sign-up URL, even if nothing is specified, so you're right this should default to `/`. Good catch.